control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -1219,16 +1210,20 @@ impl SpecOperationsHelper for VolumeSpec {
                 }),
                 _ => Ok(()),
             },
-            VolumeOperation::Publish((_, _, protocol)) => match protocol {
+            VolumeOperation::Publish((_, _, protocol, republish)) => match protocol {
                 None => Ok(()),
                 Some(protocol) => match protocol {
                     VolumeShareProtocol::Nvmf => {
                         if let Some(target) = &self.target {
-                            Err(SvcError::VolumeAlreadyPublished {
-                                vol_id: self.uuid_str(),
-                                node: target.node().to_string(),
-                                protocol: format!("{:?}", target.protocol()),
-                            })
+                            if !republish {
+                                Err(SvcError::VolumeAlreadyPublished {
+                                    vol_id: self.uuid_str(),
+                                    node: target.node().to_string(),
+                                    protocol: format!("{:?}", target.protocol()),
```

Comment:
does protocol have a to_string?

Reply:
This was exsisting code, and seems like we need to change the SvcError's parameter. Will take this up as refactor PR.